### PR TITLE
Drop Python 3.7 Support and Remove Usage of `pkg_resources`

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

`pkg_resources` was deprecated and removed in Python 3.12.

## ✔️ How to Test

You can insert a `print(VERSION)` to right after where the `VERSION` is defined. Then run any command with the Linode CLI, and you will be able to see the output.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -100,6 +100,6 @@ def bake_version(v):
         ]
     },
     data_files=get_baked_files(),
-    python_requires=">=3.7",
+    python_requires=">=3.8",
```
Comment (yec-akamai) [inline setup.py:103]: Should we add a notice somewhere saying that we are dropping support of python 3.7?

**Review by yec-akamai — APPROVED** [review]
Nice work! The CI tests are passing now.

**Review by amisiorek-akamai — APPROVED** [review]
I agree with Ye, maybe just something in the release notes?

File: setup.py (hunk 1 above)
Comment (zliang-akamai) [reply setup.py:103]: I think we can put that into the release note.

**Review by lgarber-akamai — APPROVED** [review]
Works as expected after installing Python 3.12 and running `python3 -m ensurepip --upgrade` 🎉

**Pull request merged**